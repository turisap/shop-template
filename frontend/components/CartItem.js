import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import formatMoney from '../lib/formatMoney';
import calcTotalPrice from '../lib/calcTotalPrice';
import RemoveFromCart from './RemoveFromCart';

const CartItemStyles = styled.li`
  padding: 1rem 0;
  border-bottom: 1px solid ${props => props.theme.lightgrey};
  display: grid;
  align-items: center;
  grid-template-columns: auto 1fr auto;
  img {
    margin-right: 10px;
  }
  h3,
  p {
    margin: 0;
  }
`;

const CartItem = ({cartItem}) => {
    const { price, title, image, description } = cartItem.item;
    return (
        <CartItemStyles>
            <img width={"100"} src={image} alt={title}/>
            <div className="cart-item-details">
                <h3>{title}</h3>
                <p>
                    {formatMoney(price * cartItem.quantity )}
                    { "-" }
                    <em>
                        {cartItem.quantity} &times; {formatMoney(price)} each
                    </em>
                </p>
            </div>
            <RemoveFromCart itemId={cartItem.id}/>
        </CartItemStyles>
    )
};

CartItem.propTypes = {
    cartItem : PropTypes.object.isRequired
}

export default CartItem;
