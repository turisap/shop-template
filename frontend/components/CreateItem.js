import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Router from 'next/router';
import Form from './styles/Form';
import formatMoney from '../lib/formatMoney';
import ErrorMessage from './ErrorMessage';

const CREATE_ITEM_MUTATION = gql`
    mutation CREATE_ITEM_MUTATION (
        $title : String!
        $description : String!
        $price : Int!
        $image : String,
        $largeImage : String
    ){
        createItem (
            title : $title
            description : $description
            price : $price,
            image : $image,
            largeImage : $largeImage
        )
        {
            id
        }
    }
`;

class CreateItem extends Component {
    state = {
        title : "",
        description : "",
        image : "",
        largeImage : "",
        price : 0
    };

    /**
     * This is only one handler which works with all inputs at once (lecture 17)
     * @param e
     */
    handleChange = e => {
        // destructuring the synthetic event and obtaining value as well as type and name of the input
        const { name, type, value } = e.target;
        const val = type === 'number' ? parseFloat(value) : value;
        this.setState({[name] : value});
    };

    handleSubmit = async (e, createItemFunction) => {
        e.preventDefault();
        const response = await createItemFunction();
        Router.push({
            pathname : '/item',
            query : { id : response.data.createItem.id }
        })
    };

    render() {
        return (
            <Mutation
                mutation={CREATE_ITEM_MUTATION}
                variables={this.state}
            >
                {(createItemFunction, { loading, error }) => (
                    <Form onSubmit={ e => this.handleSubmit(e, createItemFunction)}>
                        <ErrorMessage error={error}/>
                        {/** setting loading animations and disabling inputs while submitting the form **/}
                        <fieldset disabled={loading} aria-busy={loading}>
                            <label htmlFor="title">
                                Title
                                <input
                                    value={this.state.title}
                                    onChange={this.handleChange}
                                    type="text"
                                    id="title"
                                    name="title"
                                    placeholder="title"
                                    required
                                />
                            </label>
                            <label htmlFor="price">
                                Price
                                <input
                                    value={this.state.price}
                                    onChange={this.handleChange}
                                    type="number"
                                    id="price"
                                    name="price"
                                    placeholder="price"
                                    required
                                />
                            </label>
                            <label htmlFor="description">
                                Description
                                <textarea
                                    value={this.state.description}
                                    onChange={this.handleChange}
                                    id="description"
                                    name="description"
                                    placeholder="description"
                                    required
                                />
                            </label>
                        </fieldset>
                        <button>Submit</button>
                    </Form>
                )}
            </Mutation>
        );
    }
}

CreateItem.propTypes = {};

export default CreateItem;
export { CREATE_ITEM_MUTATION };
