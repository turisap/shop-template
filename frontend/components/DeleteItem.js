import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';

const DELETE_ITEM_MUTATION = gql`
    mutation DELETE_ITEM_MUTATION ($id : ID!) {
        deleteItem(id : $id){
            id
        }
    }
`;

class DeleteItem extends Component {
    render() {
        return (
            <Mutation
                mutation={DELETE_ITEM_MUTATION}
                variables={{id : this.props.id}}
            >
                {(deleteItem, {error}) => (
                    <button onClick={() => {
                        console.log(this.props.id)
                        if(confirm('Are you sure you wanna delete this item?')) deleteItem();
                    }}>{this.props.children}</button>
                )}
            </Mutation>
        );
    }
}

DeleteItem.propTypes = {};

export default DeleteItem;
