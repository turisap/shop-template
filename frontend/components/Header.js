import * as config from '../config';
import NavBar from './NavBar';

const Header = () => {
    return (
        <div>
            <div className="bar">
                <a href="">{config.applicationName}</a>
            </div>
            <NavBar/>
            <div className="sub-bar">
                <p>Search</p>
            </div>
            <div>Cart</div>
        </div>
    );
};

export default Header;
