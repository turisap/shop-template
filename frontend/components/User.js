import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import propTypes from 'prop-types';

/**here we grab an item in this query based on relationship between CartItem and Item**/
const CURRENT_USER_QUERY = gql`
  query {
    me {
      id
      email
      name
      permissions
      orders {
          id
      }
      cart {
          id
          quantity
          item {
              id
              title
              price
              image
              description
          }
      }  
    }
  }
`;

const User = props => (
    <Query {...props} query={CURRENT_USER_QUERY}>
        {payload =>  props.children(payload)}
    </Query>
);

User.propTypes = {
    children: propTypes.func.isRequired,
};

export default User;
export { CURRENT_USER_QUERY };
