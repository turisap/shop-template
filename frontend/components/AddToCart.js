import React from 'react';
import gql from 'graphql-tag';
import { Mutation } from 'react-apollo';
import { CURRENT_USER_QUERY } from "./User";

const ADD_TO_CART_MUTATION = gql`
    mutation ADD_TO_CART_MUTATION ($id : ID!) {
        addToCart(id : $id) {
            id
            quantity
        }
    }
`;

class AddToCart extends React.Component {
    render(){
        const { id } = this.props;
        return (
            <Mutation
                mutation={ADD_TO_CART_MUTATION}
                variables={{id}}
                refetchQueries={[
                    { query : CURRENT_USER_QUERY }
                ]}
            >
                {(addToCart, {loading, error}) => (
                    <button
                        disabled={loading}
                        onClick={addToCart}>
                        Add{loading && 'ing'} to Cart
                    </button>
                )}
            </Mutation>
        )
    }
}

export default AddToCart;
export { ADD_TO_CART_MUTATION };