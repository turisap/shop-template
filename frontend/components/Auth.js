import React from 'react';
import { Query } from 'react-apollo';
import { CURRENT_USER_QUERY } from "./User";
import SignIn from './SignIn';

const Authentication = props => (
    <Query query={CURRENT_USER_QUERY}>
        {({data, loading}) => {
            if(loading) return <p>Loading...</p>;
            if(!data.me) {
                return (
                    <div>
                        <p>Please sigin in before continuing</p>
                        <SignIn/>
                    </div>
                )
            }
            return props.children;
        }}
    </Query>
)

export { Authentication };