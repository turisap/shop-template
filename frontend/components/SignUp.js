import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import Error from './ErrorMessage';
import {CURRENT_USER_QUERY} from "./User";

const SIGNUP_MUTATION = gql`
    mutation SIGNUP_MUTATION($email : String!, $name : String!, $password : String!) {
        signup(email : $email, name : $name, password : $password) {
            id
            email
            name
        }
    }
`;


class SignUp extends Component {
    state = {
        name : "",
        email : "",
        password : ""
    };

    saveToState = e => {
        this.setState({
            [e.target.name] : e.target.value
        })
    };

    render() {
        return (
            <Mutation
                mutation={SIGNUP_MUTATION}
                variables={this.state}
                refetchQueries={[
                    {query : CURRENT_USER_QUERY}
                ]}
            >
                {(signUpFunction, {error, loading}) => {
                    return (
                        <Form
                            method="post"
                            onSubmit={async e => {
                                e.preventDefault();
                                await signUpFunction();
                                this.setState({name : "", email : "", password : ""});
                            }}
                        >
                            <fieldset disabled={loading} aria-busy={loading}>
                                <h2>Sign up for an account</h2>
                                <Error error={error}/>
                                <label>
                                    Email
                                    <input
                                        type="email"
                                        name="email"
                                        placeholder="Email"
                                        value={this.state.email}
                                        onChange={this.saveToState}
                                    />
                                </label>
                                <label>
                                    Name
                                    <input
                                        type="text"
                                        name="name"
                                        placeholder="Name"
                                        value={this.state.name}
                                        onChange={this.saveToState}
                                    />
                                </label>
                                <label>
                                    Password
                                    <input
                                        type="password"
                                        name="password"
                                        placeholder="Password"
                                        value={this.state.password}
                                        onChange={this.saveToState}
                                    />
                                </label>
                                <button type="submit">Sign Up</button>
                            </fieldset>
                        </Form>
                    )
                }}
            </Mutation>
        );
    }
}

SignUp.propTypes = {};

export default SignUp;
export { SIGNUP_MUTATION };
