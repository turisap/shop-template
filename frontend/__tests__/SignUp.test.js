import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { ApolloConsumer } from 'react-apollo';
import SignUp, { SIGNUP_MUTATION } from '../components/SignUp';
import { MockedProvider } from 'react-apollo/test-utils';
import { CURRENT_USER_QUERY } from "../components/User";
import { fakeUser } from "../lib/testUtils";

function type (wrapper, name, value) {
    wrapper.find(`input[name="${name}"]`).simulate('change', {
        target : { name, value }
    });
}

const me = fakeUser();
const mocks = [
    // signup mock mutation
    {
        request: {
            query: SIGNUP_MUTATION,
            variables: {
                name: me.name,
                email: me.email,
                password: 'lll',
            },
        },
        result: {
            data: {
                signup: {
                    __typename: 'User',
                    id: 'abc123',
                    email: me.email,
                    name: me.name
                },
            },
        },
    },
    // current user mock mutation
    {
        request: { query: CURRENT_USER_QUERY },
        result: {
            data: { me }
        }
    }
];
// const mocks = [
//     // signup mock mutation
//     {
//         request: {
//             query: SIGNUP_MUTATION,
//             variables: {
//                 name: me.name,
//                 email: me.email,
//                 password: 'wes',
//             },
//         },
//         result: {
//             data: {
//                 signup: {
//                     __typename: 'User',
//                     id: 'abc123',
//                     email: me.email,
//                     name: me.name,
//                 },
//             },
//         },
//     },
//     // current user query mock
//     {
//         request: { query: CURRENT_USER_QUERY },
//         result: { data: { me } },
//     },
// ];

describe('<SignUp/>', () => {
    it('renders and matches shapshot', () => {
        const wrapper = mount(<MockedProvider><SignUp/></MockedProvider>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('calls the mutation properly', async () => {
        let apolloClient;
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <ApolloConsumer>
                    {client => {
                        apolloClient = client;
                        return <SignUp />;
                    }}
                </ApolloConsumer>
            </MockedProvider>
        );
        await wait();
        wrapper.update();
        type(wrapper, 'name', me.name);
        type(wrapper, 'email', me.email);
        type(wrapper, 'password', 'lll');
        wrapper.update();
        wrapper.find('form').simulate('submit');
        await wait();
        // query the user out of the apollo client
        const user = await apolloClient.query({ query: CURRENT_USER_QUERY });
        //console.log(user);
        expect(user.data.me).toMatchObject(me);

    })
})