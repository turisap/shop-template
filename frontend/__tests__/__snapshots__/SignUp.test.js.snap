// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp/> renders and matches shapshot 1`] = `
<MockedProvider
  addTypename={true}
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "typenameDocumentCache": WeakMap {},
          "watches": Array [],
        },
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "typenameDocumentCache": WeakMap {},
            "watches": Array [],
          },
        },
        "version": "2.4.2",
        "watchQuery": [Function],
      }
    }
  >
    <SignUp>
      <Mutation
        mutation={
          Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "SIGNUP_MUTATION",
                },
                "operation": "mutation",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "email",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "email",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "name",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "password",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "password",
                            },
                          },
                        },
                      ],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "signup",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "email",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [
                  Object {
                    "defaultValue": undefined,
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NonNullType",
                      "type": Object {
                        "kind": "NamedType",
                        "name": Object {
                          "kind": "Name",
                          "value": "String",
                        },
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                    },
                  },
                  Object {
                    "defaultValue": undefined,
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NonNullType",
                      "type": Object {
                        "kind": "NamedType",
                        "name": Object {
                          "kind": "Name",
                          "value": "String",
                        },
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                    },
                  },
                  Object {
                    "defaultValue": undefined,
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NonNullType",
                      "type": Object {
                        "kind": "NamedType",
                        "name": Object {
                          "kind": "Name",
                          "value": "String",
                        },
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "password",
                      },
                    },
                  },
                ],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 223,
              "start": 0,
            },
          }
        }
        refetchQueries={
          Array [
            Object {
              "query": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "permissions",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orders",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "cart",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "quantity",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "item",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "id",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "title",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "price",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "image",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "description",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 303,
                  "start": 0,
                },
              },
            },
          ]
        }
        variables={
          Object {
            "email": "",
            "name": "",
            "password": "",
          }
        }
      >
        <Form
          method="post"
          onSubmit={[Function]}
        >
          <form
            className="Form-sc-1ns280c-0 fTwjpq"
            method="post"
            onSubmit={[Function]}
          >
            <fieldset
              aria-busy={false}
              disabled={false}
            >
              <h2>
                Sign up for an account
              </h2>
              <DisplayError
                error={Object {}}
              />
              <label>
                Email
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </label>
              <label>
                Name
                <input
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </label>
              <label>
                Password
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </label>
              <button
                type="submit"
              >
                Sign Up
              </button>
            </fieldset>
          </form>
        </Form>
      </Mutation>
    </SignUp>
  </ApolloProvider>
</MockedProvider>
`;
